lib/redmine_merge_request_links/patches/issue_query_patch.rb

```diff
@@ -0,0 +1,38 @@
+module RedmineMergeRequestLinks
+  module Patches
+    module IssueQueryPatch
+      def self.included(base)
+
+        def initialize_available_filters_with_more_filters
+          initialize_available_filters_without_more_filters
+          add_available_filter("merge_request",
+            :type => :list_optional, :values => ['open', 'merged', 'closed']
+          )
+        end
+
+        def sql_for_merge_request_field(field, operator, value, options={})
+          case operator
+          when "*", "!*"
+            "#{(operator == "*" ? "EXISTS" : "NOT EXISTS")} (SELECT 1 FROM issues_merge_requests WHERE issues_merge_requests.issue_id = issues.id)"
+          when "=", "!"
+            "EXISTS (SELECT 1 FROM issues_merge_requests JOIN merge_requests ON merge_requests.id = issues_merge_requests.merge_request_id" + 
+              " WHERE issues_merge_requests.issue_id = issues.id" + 
+              " AND merge_requests.state #{(operator == "=" ? "IN" : "NOT IN")} (" + value.collect{|val| "'#{self.class.connection.quote_string(val)}'"}.join(",") + "))"
+          end
+        end
+
+        base.class_eval do
+          unloadable
+          alias_method :initialize_available_filters_without_more_filters, :initialize_available_filters
+          alias_method :initialize_available_filters, :initialize_available_filters_with_more_filters
```

Comment:
Let's use a more specific suffix here like `_with_merge_requests` to not risk collision with other plugins.

Reply:
Done, it did in fact collide with an existing plugin https://github.com/HugoHasenbein/redmine_more_filters which I had used as inspiration.